core/matching/matchers/contains_match.go

```diff
@@ -9,10 +10,20 @@ import (
 var Contains = "contains"
 
 func ContainsMatch(match interface{}, toMatch string) bool {
-	matchStringArr, ok := match.([]string)
-	if !ok {
+	val := reflect.ValueOf(match)
+	if val.Kind() != reflect.Slice {
 		return false
 	}
+	var matchStringArr []string
+	for i := 0; i < val.Len(); i++ {
+		currentValue := val.Index(i)
+		if currentValue.Kind() == reflect.Interface {
+			matchStringArr = append(matchStringArr, currentValue.Elem().String())
+		} else {
+			matchStringArr = append(matchStringArr, currentValue.String())
+		}
+
+	}
```

Comment:
We should probably normalize these matchers into one so there is less duplication, also provide flexibility for extension, something like: 
```
"q": [
    {
        "matcher": "array",
        "options": {
          "ignoreOrder": true,
          "ignoreUnknown": true,
          "ignoreDuplicate": true
        }
        "value": [
            "access:vod",
            "order:latest",
            "profile:vd"
        ]
    }
```
what do you think?

Reply:
Yea, that works.

Reply:
@tommysitu  ll it be fine to take these options in the configuration itself?

Reply:
@tommysitu  raised PR #1033  for the same. I feel that we should keep those matchers (containsexactly, containsonly, contains) as well and let users decide which one to use based on their ease.